File: app/src/main/java/ru/profapp/RanobeReader/DAO/ChapterDao.java
package ru.profapp.RanobeReader.DAO;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ru.profapp.RanobeReader.Models.Chapter;

/**
 * Created by Ruslan on 09.02.2018.
 */

@Dao
public interface ChapterDao {
//Todo: change replace to ignore
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Chapter chapter);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(Chapter... chapters);

    @Update
    void updateAll(Chapter... chapters);
    @Update
    void update(Chapter chapter);

    @Delete
    void delete(Chapter chapter);

    @Query("SELECT * FROM chapter WHERE Url = :ChapterUrl")
    Chapter getByChapterUrl(String ChapterUrl);


    @Query("SELECT * FROM chapter")
    List<Chapter> getAllChapters();

    @Query("SELECT * FROM chapter WHERE RanobeUrl IS :RanobeUrl order by chapter.`Index` Asc")
    List<Chapter> getChaptersForRanobe(String RanobeUrl);

}

File: app/src/main/java/ru/profapp/RanobeReader/DAO/ChapterDao.kt
package ru.profapp.RanobeReader.DAO

import androidx.room.*
import ru.profapp.RanobeReader.Models.Chapter

/**
 * Created by Ruslan on 09.02.2018.
 */

@Dao
interface ChapterDao {


    @Query("SELECT * FROM chapter")
    fun allChapters(): List<Chapter>


    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(chapter: Chapter)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(vararg chapters: Chapter)

    @Update
    fun updateAll(vararg chapters: Chapter)

    @Update
    fun update(chapter: Chapter)

    @Delete
    fun delete(chapter: Chapter)

    @Query("SELECT * FROM chapter WHERE Url = :ChapterUrl")
    fun getByChapterUrl(ChapterUrl: String): Chapter

    @Query("SELECT * FROM chapter WHERE RanobeUrl IS :RanobeUrl order by chapter.`Index` Asc")
    fun getChaptersForRanobe(RanobeUrl: String): List<Chapter>

}
